src/rust/ensogl/lib/core/src/display/object/class.rs

```diff
@@ -573,6 +573,16 @@ pub struct Instance<Host=Scene> {
     rc : Rc<Model<Host>>
 }
 
+impl<Host> Drop for Instance<Host> {
+    fn drop(&mut self) {
+        // If this is the last reference, remove it from its parent. If there is a parent, then this
+        // will create a new reference which will be dropped on the parent's next update.
+        if Rc::strong_count(&self.rc) == 1 {
+            self.unset_parent();
+        }
+    }
+}
+
```

Comment:
Is that needed?
When we destroy Instance, we also destroy the parent_biding, which should remove the node from the parent upon dropping.

Reply:
It is needed, so we do not drop the object immediately and can call `on_hide` when the parent gets updated. Should I clarify this in the comment?

Reply:
Ach, now I see. ParentBind will not keep the instance, and the parent won't be able to upgrade Weak.

Reply:
Right